packages/subgraph/src/QFIMapping.ts

```diff
@@ -0,0 +1,572 @@
+    const qfiAddress = event.address
+    const qfiContract = QFIContract.bind(qfiAddress)
+    const qfiId = qfiAddress.toHexString()
+    const qfi = new QFISchema(qfiId)
+
+    qfi.stateAqAddress = event.params._stateAq
+    qfi.pollFactoryAddress = event.params._pollFactory
+    qfi.initialVoiceCreditProxyAddress = event.params._initialVoiceCreditProxy
+    qfi.signUpGatekeeperAddress = event.params._signUpGatekeeper
+    qfi.signUpTimestamp = event.params._timestamp
+    qfi.numSignUps = new BigInt(0)
+    qfi.nextGrantRoundId = new BigInt(0)
+    qfi.createdAt = timestamp
+    qfi.lastUpdatedAt = timestamp
+
+    // External call to contract instance (one time event - good trade-off for sync time).
+    qfi.stateTreeDepth = qfiContract.stateTreeDepth()
+    qfi.numSignUps = qfiContract.numSignUps()
+
+    qfi.save()
+}
+
+/**
+ * Handle a new QFI instance deploy.
+ * @param event Ethereum event emitted when a new QFI instance is deployed.
+ */
+export function handleQfiDeployed(event: QfiDeployed): void {
+    log.debug(`QFI Deployed event block: {}`, [event.block.number.toString()])
+
+    // Get the QFI/MACI instance.
+    const qfiAddress = event.address
+    const qfiId = qfiAddress.toHexString()
+    const qfi = new QFISchema(qfiId)
+
+    if (qfi !== null) {
+        qfi.grantRoundFactoryAddress = event.params._grantRoundFactory
+        qfi.nativeERC20TokenAddress = event.params._nativeToken
+        qfi.voiceCreditFactor = event.params._voiceCreditFactor
+        qfi.currentStage = "NotInitialized"
+
+        qfi.save()
+
+        log.info("QFI has been correctly deployed!", [])
+    } else {
+        log.error(`QFI entity not found!`, [])
+    }
+}
+
+/**
+ * Handle the initialization of a smart contract based on MACI.
+ * @param event Ethereum event emitted when initializing a smart contract based on MACI.
+ */
+export function handleInitMaci(event: Init): void {
+    log.debug(`Init MACI event block: {}`, [event.block.number.toString()])
+
+    // Get the QFI/MACI instance.
+    const qfiAddress = event.address
+    const qfiId = qfiAddress.toHexString()
+    const qfi = new QFISchema(qfiId)
+
+    if (qfi !== null) {
+        qfi.vkRegistryAddress = event.params._vkRegistry
+        qfi.messageAqFactoryPollAddress = event.params._messageAqFactory
+        qfi.isInitialized = true
+
+        qfi.save()
+    }
+}
+
+/**
+ * Handle the initialization of the QFI smart contract.
+ * @param event Ethereum event emitted when initializing the QFI smart contract.
+ */
+export function handleInitQfi(event: InitQfi): void {
+    log.debug(`Init QFI event block: {}`, [event.block.number.toString()])
+
+    // Get the QFI/MACI instance.
+    const qfiAddress = event.address
+    const qfiId = qfiAddress.toHexString()
+    const qfi = new QFISchema(qfiId)
+
+    if (qfi !== null) {
+        qfi.messageAqFactoryGrantRoundAddress = event.params._messageAqFactoryGrantRounds
+        qfi.currentStage = "WaitingForSignupsAndTopups"
```

Comment:
I created a Stage enum but it seems that it is not possible to use it directly as any type in the mappings (also, I can't see anything related to my Stage enum in the `generated/` folder). As stated in [TheGraph Docs](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#enums) _"Once the enum is defined in the schema, you can use the string representation of the enum value to set an enum field on an entity."_

Maybe a good workaround can consist in adding a parameter to the events where the current stage value changes and then set the value in the mapping using that event parameter. Do you think that's a good way to solve it? It should be efficient enough because we are adding a small `uint8` parameter.